backend/corpora/dataset_processing/download.py

```diff
@@ -74,6 +75,27 @@ def updater(processing_status_uuid: str, tracker: ProgressTracker, frequency: fl
     """
 
     def _update():
+        with db_session_manager(commit=True) as db:
+            try:
+                curr_status = db.get(DbDatasetProcessingStatus, processing_status_uuid).upload_status
+                if curr_status is UploadStatus.CANCEL_PENDING:
+                    logger.info(f"cancelling the upload for {curr_status.dataset.id}")
+                    # set db to cancelled
+                    status = {
+                        DbDatasetProcessingStatus.upload_progress: 0,
+                        DbDatasetProcessingStatus.upload_status: UploadStatus.CANCELED,
+                        DbDatasetProcessingStatus.upload_message: "Canceled by user",
+                    }
+
+                    dataset = Dataset.get(curr_status.dataset.id)
```

Comment:
remove line 90-91. The database will be updated at the bottom of this function on line 122.

Reply:
wont status get overwritten by one of the following if statements?

Reply:
you make a good point. You can shorten the scope of the db_session_manager context to only retrieve the curr_status. Then everything will be at the same level as the main if statement. Then add `if curr_status is UploadStatus.CANCEL_PENDING:` as the first if statement. I think this will also resolve your need for the `try except` block.